client/src/components/Inputs/InputField/index.js

```diff
@@ -0,0 +1,129 @@
+
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import InputAdornment from "@material-ui/core/InputAdornment";
+import IconButton from "@material-ui/core/IconButton";
+
+import * as T from "../../Typograpy";
+
+import Icon from "../../Icon";
+import useStyles from "../style";
+
+const InputField = (props) => {
+  const {
+    // input type: number, text, textarea, email or password
+    type = "text",
+    // placeholder text
+    placeholder,
+    // label above box
+    label,
+    // error for validation render
+    error,
+    // normal, large, xLarge
+    size = "normal",
+    // material ui id if needed
+    id = "outlined-basic",
+    // get functions to set value and value itself from parent component
+    setValue,
+    value,
+  } = props;
+
+  const classes = useStyles(props);
+
+  //  password show / hide UI
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleClickShowPassword = () => {
+    setShowPassword(!showPassword);
+  };
+
+  const handleMouseDownPassword = (event) => {
+    event.preventDefault();
+  };
+
+  // renders typography for labels above box
+  const renderLabel = {
+    normal: <T.Body18B className={classes.textInputLabel}>{label}</T.Body18B>,
+    large: <T.Body20B className={classes.textInputLabel}>{label}</T.Body20B>,
+    largeOptional: (
+      <T.Body20B className={classes.textInputLabel}>
+        {label}
+        {/* optional brackets render */}
+        <T.Body14R
+          style={{
+            marginBottom: "1rem",
+            marginLeft: "0.3rem",
+          }}
+          variant="brackets"
+        >
+          (optional)
+        </T.Body14R>
+      </T.Body20B>
+    ),
+    xLarge: <T.Body30B className={classes.textInputLabel}>{label}</T.Body30B>,
+  };
+
+  // selects input type
+  const inputTypes = {
+    text: "text",
+    password: showPassword === true ? "text" : "password",
+    email: "email",
+    number: "number",
+    textarea: "textarea",
+  };
+
+  const textfieldTypes = ["text", "textarea", "email", "password", "number"];
+
+  const renderTextField = (
+    <TextField
+      error={!!error}
+      type={inputTypes[type]}
+      value={value}
+      onChange={setValue}
+      id={id}
+      variant="outlined"
+      multiline={type === "textarea"}
+      rows={type === "textarea" && 4}
+      placeholder={placeholder}
+      InputProps={{
+        classes: {
+          root: classes.input,
+          input: classes.inputPlaceholder,
+        },
+        // deals with password visibility
+        endAdornment: type === "password" && (
+          <InputAdornment position="end">
+            <IconButton
+              onClick={handleClickShowPassword}
+              onMouseDown={handleMouseDownPassword}
+              aria-label="toggle password visibility"
+            >
+              <Icon width="40px" icon="eye" />
+            </IconButton>
+          </InputAdornment>
+        ),
+      }}
+    />
```

Comment:
i should be able to pass additional props like name, id, onfocus ..
```suggestion
    />
```
```suggestion
{...props}
    />
```